services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3532,17 +3531,27 @@ private boolean hasFatalDataValidationError(PushMonitor pushMonitor, String topi
     }
   }
 
-  private void safeDeleteRTTopic(String clusterName, String storeName, Store store) {
-    // Perform RT cleanup checks for batch only store that used to be hybrid. Check the local versions first
-    // to see if any version is still using RT and then also check other fabrics before deleting the RT. Since
-    // we perform this check everytime when a store version is deleted we can afford to do best effort
+  private void safeDeleteRTTopic(String clusterName, String storeName) {
+    if (isRTTopicDeletionPermittedByAllControllers(clusterName, storeName)) {
+      String rtTopicToDelete = Version.composeRealTimeTopic(storeName);
+      deleteRTTopicFromAllFabrics(rtTopicToDelete, clusterName);
+    }
+    // Check if there is incremental push topic exist. If yes, delete it and send out to let other controller to
+    // delete it.
+    String incrementalPushRTTopicToDelete = Version.composeSeparateRealTimeTopic(storeName);
+    if (getTopicManager().containsTopic(pubSubTopicRepository.getTopic(incrementalPushRTTopicToDelete))) {
```

Comment:
The semantic of this refactor is actually different now. I think what you actually want here is save the result of `isRTTopicDeletionPermittedByAllControllers` to a boolean like `rtDeletionPermitted` and then here you want to have both condition to be true before deleting, i.e.
```
if (rtDeletionPermitted && getTopicManager().containsTopic(pubSubTopicRepository.getTopic(incrementalPushRTTopicToDelete)))
```
Otherwise it's not a safe delete

Reply:
got it